### PR TITLE
feat(controllers): add new controller mapping for Numark NS6II

### PR DESCRIPTION
**Pull request by Swiftb0y**
~~Manual still TODO~~
[manual PR](https://github.com/mixxxdj/manual/pull/687)

### CODE REVIEW

**Comment by Swiftb0y** [comment]
Ah yes, good old stale. I have had and used this controller for about 5 years now, one day I'll get around to polishing the mapping enough for merge.

Still, a surface-level review in the meantime would be appreciated.

**Review by acolombier — COMMENTED** [review]
Hard to review without the device to test. but it is looking good overall. Also I'm sure that if you've been using it this long, you probably had a chance to polish most of the small bug lying around :) 

Some comment cruft here and there that might be worth removing or documenting if useful for alternate behaviours (note that if the latter, you might want to consider the controller settings).
Did you have a chance to prepare the manual for that?

**Comment by Swiftb0y** [comment]
I'll try to give this another look and ping you when I think its ready. Thanks for the review.

**Comment by Swiftb0y** [comment]
@acolombier please re-review ;)
I'm mostly satisfied with the mapping (there are a couple TODO's but I'm really not in the mood to implement those).
I'll focus on the manual page next.

File: res/controllers/midi-controller-api.d.ts
Comment (JoergAtGithub) [inline res/controllers/midi-controller-api.d.ts:3]: I can't spot the place, where this type is defined in our API. Is this a mapping specific type, that doesn't belong to the API?

Comment (Swiftb0y) [reply res/controllers/midi-controller-api.d.ts:3]: Its the type our input handlers are expected to be https://github.com/mixxxdj/mixxx/wiki/midi%20scripting#midi-input-handling-functions

**Comment by Swiftb0y** [comment]
friendly ping... ;)

File: res/controllers/Numark-NS6II-scripts.js
Comment (acolombier) [inline res/controllers/Numark-NS6II-scripts.js:710]: Could it be worth adding a dynamic setting for that?

File: res/controllers/Numark-NS6II-scripts.js
Comment (acolombier) [inline res/controllers/Numark-NS6II-scripts.js:397]: Can this be removed as this is redundant with line 409/414?

File: res/controllers/Numark-NS6II-scripts.js
Comment (acolombier) [inline res/controllers/Numark-NS6II-scripts.js:738]: Is this outstanding? Or should it be removed?

File: res/controllers/Numark-NS6II-scripts.js
Comment (acolombier) [inline res/controllers/Numark-NS6II-scripts.js:750]: Ditto

File: res/controllers/Numark-NS6II-scripts.js
Comment (Swiftb0y) [reply res/controllers/Numark-NS6II-scripts.js:710]: yes, I'll add that as a follow up, but I wanted to keep this branch targetted at 2.4

File: res/controllers/Numark-NS6II-scripts.js
Comment (Swiftb0y) [reply res/controllers/Numark-NS6II-scripts.js:738]: nah, its technically outstanding, but I really didn't want to bother implementing that because `components.HotcueButton` is not setup for that.

File: res/controllers/Numark-NS6II-scripts.js
Comment (Swiftb0y) [reply res/controllers/Numark-NS6II-scripts.js:397]: technically yes, but I would've preferred this to stay here instead of having it buried in the implementation.

**Comment by acolombier** [comment]
I had a thorough look at that `eslint` failing, I really can't find what is wrong here. I reproduced to the exact everything the pipeline does (merge commit and container) and yet, everything is fine on my local. I'm running out of idea currently, but I'll try to think of something else.
Otherwise, since `eslint` is giving me a green light locally, I'd say we could ignore the CI failing and merge it, wdyt?

**Comment by Swiftb0y** [comment]
well the build failure is fortunately unrelated and fixed with #13904. I'll investigate whats wrong with pre-commit.

**Comment by Swiftb0y** [comment]
now depends on #13913

**Comment by Swiftb0y** [comment]
How do we want to proceed? 2.4 is EOL, I can rebase on 2.5 and add appropriate controller settings. Anything else to do to get this merged some time? The hardware I have is slowly dying so I won't be able to work on this for very long. It would be a shame if I went trough the entire controller without the mapping ever making it in.

**Comment by daschuer** [comment]
You can just change the release branch via "edit" above. Alternative is to merge it to the 2.4 branch and than merge 2.4 to 2.5.

**Comment by Swiftb0y** [comment]
right, but there is little point when nobody is bothering to review or press merge.

**Comment by Swiftb0y** [comment]
added the 2.5 controller settings feature. Would you be so kind and do another review round @acolombier? Thank you.

**Review by acolombier — APPROVED** [review]
LGTM! Thanks for you work here. I'll let @ronso0 hit merge once the manual PR is  being merged too :)

**Comment by Swiftb0y** [comment]
Did a final rebase just in case. Should be good to go.

**Comment by ronso0** [comment]
Alright, thanks!
waiting for CI...

**Comment by Swiftb0y** [comment]
Passed. lets go.

**Comment by Swiftb0y** [comment]
Thank you both @acolombier and @ronso0 for pulling through and finally getting this merged. I'm very relieved to be able to focus on the Reloop Ready mapping next #4513 and afterwards on my Traktor S8.

**Pull request merged**
